util/test/checkChplInstall

```diff
@@ -23,213 +24,146 @@
-    log_debug "${chpl_dir} does not exist. Creating it."
-    mkdir -p $chpl_dir || { log_error "Failed to create ${chpl_dir}." && exit 10 ; }
-fi
+# Log file for compile output
+TEST_COMP_OUT=${TEST_JOB}.comp.out
 
-TMP_TEST_DIR=$(mktemp -d ${chpl_dir}/chapel-test-XXXXX)
-log_debug "TMP_TEST_DIR: ${TMP_TEST_DIR}"
+# Necessary to suppress warnings when WARNINGS=1 (e.g. nightly build settings)
+COMP_FLAGS="--cc-warnings"
 
-# Ensure the tmp dir is deleted when script exits.
+# Compile test job into temporary directory
+${CHPL_BIN} ${TEST_DIR}/${TEST_JOB}.chpl ${COMP_FLAGS} -o ${TMP_TEST_DIR}/${TEST_JOB} > ${TMP_TEST_DIR}/${TEST_COMP_OUT} 2>&1
+
+# Check that compile was successful
+COMPILE_STATUS=$?
+if [ ${COMPILE_STATUS} -ne 0 ]; then
+    err "Test job failed to compile, Chapel is not installed correctly"
+    echo "Compilation output:"
+    cat ${TMP_TEST_DIR}/${TEST_COMP_OUT}
+    exit 1
+else
+    echo "Test job compiled into ${TMP_TEST_DIR}/${TEST_JOB}"
+fi
+
+# Cleanup the tmp directory whenever exit is invoked
 function cleanup_tmp_dir()
 {
-    # Leave the directory when using --debug.
-    if [ -z "${debug_chpl_test}" ] ; then
-        rm -rf $TMP_TEST_DIR || :
-    else
-        log_debug "Leaving tmp test dir: ${TMP_TEST_DIR}"
-    fi
+    rm -rf ${TMP_TEST_DIR}
 }
 trap cleanup_tmp_dir EXIT
 
-test_output_log=$TMP_TEST_DIR/test_output.log
-cp $TESTS $TMP_TEST_DIR/
+# Find number of locales and .good file
+if [ ${chpl_comm} == "none" ]; then
+    NUMLOCALES=1
+    GOOD=${TEST_DIR}/${TEST_JOB}.comm-none.good
+else
+    NUMLOCALES="$(cat ${TEST_DIR}/NUMLOCALES)"
+    GOOD=${TEST_DIR}/${TEST_JOB}.good
+fi
+
+# Check for valid launchers
+if [ ${chpl_launcher} == "slurm-srun" -o ${chpl_launcher} == "amudprun" -o ${chpl_launcher} == "none" ]; then
+    echo "Launcher, \$CHPL_LAUNCHER = ${chpl_launcher}, is compatible with test script"
+else
+    echo "This script is not compatible with the \$CHPL_LAUNCHER value of ${chpl_launcher}"
+    echo "This does not necessarily indicate that your Chapel installation is incorrect"
+    echo "See \$CHPL_HOME/doc/README.launcher for more information on how to manually run a Chapel program"
+    exit 10
+fi
 
-# Move to the temp directory and run the tests!
+TEST_EXEC_OUT=${TEST_JOB}.exec.out
+
+# Run compiled binary with parsed execution options
 (
-    cd $TMP_TEST_DIR
-    export CHPL_TEST_UTIL_DIR=$CHPL_HOME/util
+    cd ${TMP_TEST_DIR}
 
-    # Run the tests.
-    echo "Running the ${test_name} tests found in: $TEST_DIR"
+    EXEC_OPTS="$(cat ${TEST_DIR}/${TEST_JOB}.execopts)"
 
-    $CHPL_HOME/util/start_test --no-chpl-home-warn --progress $CHPL_START_TEST_ARGS *.chpl > $test_output_log
+    echo "Running test job"
+    ./${TEST_JOB} -nl${NUMLOCALES} ${EXEC_OPTS} > ${TEST_EXEC_OUT} 2>&1
+    echo "Test job complete"
 
-    test_status=$?
-    log_debug "start_test exit code: ${test_status}"
-)
+    # Quick test to verify fail case
+    #echo "this should break the test" >> ${TEST_EXEC_OUT}
```

Comment:
I wouldn't keep this comment in the committed version. It's simple enough that somebody could add it back in if needed.